src/codegen.cpp

```diff
@@ -4362,6 +4362,17 @@ static jl_cgval_t emit_invoke(jl_codectx_t &ctx, jl_expr_t *ex, jl_value_t *rt)
     return emit_invoke(ctx, lival, argv.data(), nargs, rt);
 }
 
+STATIC_INLINE _jl_specsig_flags_t wait_for_code_instance(jl_code_instance_t *ci)
+{
+    _jl_specsig_flags_t specsig;
+    specsig.bits = jl_atomic_load_acquire(&ci->specsigflags);
+    while (!specsig.flags.specptr_matches_invokeptr) {
+        jl_cpu_pause();
+        specsig.bits = jl_atomic_load_acquire(&ci->specsigflags);
+    }
+    return specsig;
```

Comment:
style nit: this is a somewhat more canonical style for our code for this pattern. It is the same length, but removes a duplicate line:
```suggestion
    while (1) {
        _jl_specsig_flags_t specsig;
        specsig.bits = jl_atomic_load_acquire(&ci->specsigflags);
        if (specsig.flags.specptr_matches_invokeptr)
            return specsig;
        jl_cpu_pause();
    }
```